foxpuppet/foxpuppet.py

```diff
@@ -2,13 +2,22 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from .browser.window import BrowserWindow
-from .windows import Windows
+from .decorators import use_class_as_property
 
 
 class FoxPuppet(object):
 
     def __init__(self, selenium):
-        self.selenium = selenium
-        self.browser = BrowserWindow(selenium)
-        self.windows = Windows(selenium)
+        self._selenium = selenium
+
+    @property
+    def selenium(self):
+        return self._selenium
+
+    @use_class_as_property('api.windows.Windows')
+    def windows(self):
+        pass
+
+    @use_class_as_property('api.browser.window.BrowserWindow')
```

Comment:
I think we want to drop this decorator completely. It wasn't such a great idea to use in Puppeteer either.